src/server/accessors/ModifyCreator.ts

```diff
@@ -45,14 +54,24 @@ export class ModifyCreator implements IModifyCreator {
         return new RoomBuilder(data);
     }
 
-    public finish(builder: IMessageBuilder | ILivechatMessageBuilder | IRoomBuilder): Promise<string> {
+    public startUser(data?: IUserCreation): IUserBuilder {
+        if (data) {
+            delete data.id;
+        }
+
+        return new UserBuilder(data);
+    }
+
+    public finish(builder: IMessageBuilder | ILivechatMessageBuilder | IRoomBuilder | IUserBuilder): Promise<string> {
```

Comment:
This keeps haunting us... I wonder if it isn't time we stopped following this start/finish pattern 🤔

Reply:
Should we also remove the `IUserBuilder` and `UserBuilder` from the code base?

Reply:
No, I don't think so. I think for now we just enable Apps to call a `modify.getCreator().finishUser(builder: IUserBuilder)` - this is going to make it possible to stop adding types on the `finish` method

Reply:
Do you mean make different `finish*()` methods standalone?

Reply:
@d-gubert One thing to note is that if you go that route, you will have to introduce a breaking change and all of the previous Apps will no longer work on new installations. The reason I say you will "have to" is because if you have two places for them to do it then it's not a clear and concise. If you don't want this method to be so long, then create a specific type above for it like:

`type BuilderFinisher = IMessageBuilder | ILivechatMessageBuilder | IRoomBuilder | IUserBuilder;`

Then in the finish method do `public finish(builder: BuilderFinisher): Promise<string>`

Reply:
@graywolf336 I think that could be a solution for now :)

But I don't think that introducing new ways of achieving things in the engine should require breaking changes - we can always deprecate APIs and keep them around for compatibility reasons until the time comes for the next major version. Even if it means having "two places for them to do it then it's not clear and concise". This could be solved by improving our docs effort (which we definitely have to)

> Do you mean make different `finish*()` methods standalone?

@lolimay that's what I was thinking, but let's maybe leave this change for another PR